Move particle parameters into their own top-level subsection

gassmoeller opened the pull request:
This PR extends #5381 to perform one of the updates we discussed for 3.0: To put the particle input parameters in their own top level subsection and disentangle the use of "Number of particles" parameter between different particle generators (move it into specific subsections). 

This is WIP, because I need to check the tests and also want to do some upgrades to the older update scripts before merging this one (once this is merged many users will have to use the update script, and some of our older update scripts dont work anymore, we should remove them).

MFraters reviewed (commented):
This is great, thanks! I have a few small comments. You do need to fix some tests, but otherwise it looks good to me. 

@tjhei, do you also want to take a look at this?

gassmoeller commented:
I addressed your comments and fixed the tests.

> interesting, this line is removed, but not added later. This is fine in this case because the generator doesn't need t. Does this mean the script know about it not being used somehow, or is that is case that could go wrong?

Yes, the update script handles `Number of particles` the following way:
  - if a generator was manually selected that uses the parameter, move it into the section of that generator
  - if any other generator was selected delete the parameter, since it is currently unused and just confusing
  - if no generator was selected, move it into the subsection of the default generator (which uses the parameter)

MFraters reviewed (approved):
Cool! This looks ready to merge to me.

tjhei commented on doc/modules/changes/20230910_gassmoeller:
```diff
@@ -0,0 +1,9 @@
+at the top level. This is to make it more clear that
+particles are not simply a postprocessing feature.
+All included input files have been adjusted and the reformatting
+script has been adjusted. Users will have to apply the reformatting
```
can you link to the documentation and/or name the script? (can be a separate PR)
